perimetre_asa/asaperimetre.py

```diff
@@ -29,16 +29,18 @@
 from qgis.core import QgsApplication
 from qgis.utils import iface
 from qgis.PyQt.QtCore import Qt
+from qgis.PyQt.QtWidgets import QMenu
 
 from .dock import AsaPerimetreDock
 from .processing.provider import AsaPerimetreProvider
 
 
 class AsaPerimetre:
 
-    def __init__(self):
+    def __init__(self, iface):
```

Comment:
This is doing a "shadow" variable `iface` because `iface` is already a existing variable in the context, L30.
A Python IDE is raising a warning for that.
Why not using the existing iface ?